docs/csharp/language-reference/keywords/fixed-statement.md

```diff
@@ -10,48 +10,56 @@ f1_keywords:
+
+[!code-csharp[Accessing fixed memory](../../../../samples/snippets/csharp/keywords/FixedKeywordExamples.cs#1)]
+
+You can initialize a pointer by using an array, a string, a fixed-size buffer, or the address of a variable. The following example illustrates the use of variable addresses, arrays, and strings. For more information about fixed-size buffers, see [Fixed Size Buffers](../../programming-guide/unsafe-code-pointers/fixed-size-buffers.md).
+
+[!code-csharp[Initializing fixed size buffers](../../../../samples/snippets/csharp/keywords/FixedKeywordExamples.cs#2)]
+
+Multiple pointers can be initialized in one statement if they are all the same type:
+
 ```csharp
-fixed (byte* ps = srcarray, pd = dstarray) {...}  
+fixed (byte* ps = srcarray, pd = dstarray) {...}
 ```
-  
- To initialize pointers of different types, simply nest `fixed` statements, as shown in the following example.  
-  
- [!code-csharp[csrefKeywordsFixedLock#3](../../../csharp/language-reference/keywords/codesnippet/CSharp/fixed-statement_3.cs)]  
-  
- After the code in the statement is executed, any pinned variables are unpinned and subject to garbage collection. Therefore, do not point to those variables outside the `fixed` statement.  
-  
+
+To initialize pointers of different types, simply nest `fixed` statements, as shown in the following example.
+
+[!code-csharp[Initializing multiple pointers](../../../../samples/snippets/csharp/keywords/FixedKeywordExamples.cs#3)]
+
+After the code in the statement is executed, any pinned variables are unpinned and subject to garbage collection. Therefore, do not point to those variables outside the `fixed` statement. The variables declared in the `fixed` statement are scoped to that statement, making this easier:
+
+```csharp
+fixed (byte* ps = srcarray, pd = dstarray) 
+{
+   ...
+}
+// ps and pd are no longer in scope here.
+```
+
 > [!NOTE]
->  Pointers initialized in fixed statements cannot be modified.  
-  
- In unsafe mode, you can allocate memory on the stack, where it is not subject to garbage collection and therefore does not need to be pinned. For more information, see [stackalloc](../../../csharp/language-reference/keywords/stackalloc.md).  
-  
-## Example  
- [!code-csharp[csrefKeywordsFixedLock#4](../../../csharp/language-reference/keywords/codesnippet/CSharp/fixed-statement_4.cs)]  
-  
-## C# Language Specification  
- [!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]  
-  
-## See Also  
- [C# Reference](../../../csharp/language-reference/index.md)  
- [C# Programming Guide](../../../csharp/programming-guide/index.md)  
- [C# Keywords](../../../csharp/language-reference/keywords/index.md)  
- [unsafe](../../../csharp/language-reference/keywords/unsafe.md)  
- [Fixed Size Buffers](../../../csharp/programming-guide/unsafe-code-pointers/fixed-size-buffers.md)
+> Pointers initialized in fixed statements cannot be modified.
```

Comment:
I'm not quite sure I understand what this means. You can, for instance, get the pointer to the first element of an array, then increment the pointer to iterate the array? Isn't that modifying the pointer?

Reply:
I had to look this up.  I re-worded it, and added a short sample.